regular_expressions_basics/regular_expressions_basics.md

```diff
@@ -0,0 +1,338 @@
+
+`. ^ $ * + ? { } [ ] \ | ( ) - /`
+
+If you want to refer to one of these special characters in regex (for example, if your pattern includes square brackets or parentheses or periods), you can add a backslash (` \ `) just before the symbol. The backslash in regex does a couple of different things, but here, the use of a backslash indicates "I mean this character literally, I'm not using this character's special powers", and this is called **escaping** or an **escape sequence**.
+
+<div class = "important">
+<b style="color: rgb(var(--color-highlight));">Important note</b><br>
+
+Sometimes the hyphen (`-`) and the forward slash (`/`) are not included in lists of metacharacters, because these two characters only have a special function in certain contexts, not all the time.  We include them in our list of metacharacters here because they do *sometimes* have a special function, and you can put a backslash in front of them to indicate you mean the literal symbol and not its special meaning.  
+
+In most uses of regex, if you're not sure if a (non-alphanumeric) character has a special meaning or not, you can put a backslash in front of it just in case.  So, for example, the hash symbol `#` doesn't have a special function in regex.  But if you weren't sure, you can usually add the backslash, giving `\#` to indicate `#`.  It's not required, but most regular expression parsers will realize what you mean.
+
+</div>
+
+* To indicate a question mark in your pattern, you'd escape it like this: `\?`.
+* A period would similarly use the backslash: `\.`
+* Want to indicate a left curly brace? `\{`
+* Vertical pipe? `\|`
+* A backslash itself?  `\\ `
+* A hyphen/dash? `\-`
+* A percent sign? Either `%` by itself, because it's not a metacharacter, or `\%` because you weren't sure and added the escape sequence just in case.
+
+<div class = "important">
+<b style="color: rgb(var(--color-highlight));">Important note</b><br>
+
+Importantly, don't try to use the backslash this way in front of letters or numbers. If you add `\d` to your pattern, for example, that doesn't mean "the literal letter d", it means "any digit 0-9".  If you add `\7` to your regular expression, you'll get an error -- `\7` is bad syntax for regular expressions. 
+
+We'll talk about other uses of backslash in an upcoming section.
+
+</div>
+
+Combining letters, numbers, spaces, symbols, and escape sequences
+----
+
+You can combine escaped characters and normal characters.  For example, if your pattern includes ".co.nz #5", you would express that as `\.co\.nz #5`.
+
+### Quiz 1: Specific Characters
+
+Which regular expression below will match the text "25% (20) claim:"?  Check all that apply.
+
+[[X]] `25\% \(20\) claim\:`
+[[X]] `25% \(20\) claim:`
+[[ ]] `25% (20) claim:`
+[[ ]] `25\% (20) claim\:`
+[[?]] Hint: Feeling stuck?  Try using [Regex 101](https://www.regex101.com) and plug in the regex patterns given in the answers as well as the test string posed in the question.
```

Comment:
Because regex101 is now introduced in a separate module, maybe add a sentence to this hint about "trying out this online tool for checking regular expressions" here.

Reply:
Again, I'm thinking that not everyone will take each module, so a bit of specificity here is warranted.  Thoughts?

Reply:
I now retract this comment, in light of your responses the the other regex 101 comment and my response to that :)